db/db_impl/db_impl.cc

```diff
@@ -4909,7 +4909,14 @@ Status DBImpl::WriteOptionsFile(bool need_mutex_lock,
 
   if (s.ok()) {
     s = RenameTempFileToOptionsFile(file_name);
+  } else {
```

Comment:
So just to confirm, It won't consider failure in renaming the file as the deletion check is inside the else statement?

Reply:
Good point. Its the same situation if rename fails - the temp OPTIONS file is not deleted. It needs to be handled.